src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubConfiguration.java

```diff
@@ -55,17 +59,59 @@ public GitHubConfiguration() {
     }
 
     @NonNull
-    public List<Endpoint> getEndpoints() {
+    public synchronized List<Endpoint> getEndpoints() {
```

Comment:
We are using copy on write semantics. We replace the entire list every time in setEndpoints